Added Ayu Light and Ayu Mirage

LoveSponge opened the pull request:
This is an initial commit and my first time with an emacs theme, let alone a doom-emacs theme, feel free to be harsh in your comments.

hlissner commented:
Could you include screenshots?

LoveSponge commented:
Sure thing - #501

ema2159 commented:
@LoveSponge that's not what we mean :sweat_smile:. A screenshot like in #447.

ema2159 commented:
What we mean is that you have to paste a couple of screenshots here in the thread in order to have a quick peek of the themes.
The other PR is also ok because those screenshots will also be merged when the themes are included.

Emiller88 commented on themes/doom-ayu-light-theme.el:
```diff
@@ -0,0 +1,264 @@
+;; mirage colours:
+;;  [path]      [colour] [light]                   [mirage]           [dark]
+;; common:
+;;   accent:    orange   ff9940                     ffcc66            e6b450
+;;   bg:        black    fafafa                     1f2430            0a0e14
+;;   fg:        grey     575f66                     cbccc6            b3b1ad
+;;   ui:        grey     ba9199                     707abc            4d5566
+;; syntax:
+;;   tag:       cyan     55b4d4                     5ccfe6            39bae6
+;;   func:      yellow   f2ae49                     ffd580            ffb454
+;;   entity:    blue     399ee6                     73d0ff            59c2ff
+;;   string:    green    86b300                     bae67e            c2d94c
+;;   regexp:    teal     4cbf99                     95e6cb            95e6cb
+;;   markup:    red      f07171                     f28779            f07178
+;;   keyword:   orange?  fa8d3e                     ffa759            ff8f40
+;;   special:   yellow?  e6ba7e                     ffe6b3            e6b673
+;;   comment:   grey     abb0b6                     5c6773            626a73
+;;   constant:  magenta  a37acc                     d4bfff            ffee99
+;;   operator:  orange?  ed9366                     f29e74            f29668
+;;   error:     red      f51818                     ff3333            ff3333
+;; ui:
+;;   line:               ui.alpha(0.1)              bg.darken(0.15)   bg.darken(0.2)
+;;   panel:
+;;     bg:               bg.brighten(0.1)           bg.brighten(0.1)  bg.brighten(0.05)
+;;     shadow:           fg.darken(0.5).alpha(0.25) bg.darken(0.3)    bg.darken(0.2)
+;;     border:           bg.darken(0.2)             bg.darken(0.4)    bg.darken(0.8)
+;;   gutter:
+;;     normal:           ui.alpha(0.4)              ui.alpha(0.4)     ui.alpha(0.6)
+;;     active:           ui.alpha(0.8)              ui.alpha(0.8)     ui.alpha(0.9)
+;;   selection:
+;;     bg:               2ea8e6.fade(0.7)           7399e6.fade(0.87) modified.fade(0.87)
+;;     inactive:         000000.fade(0.87)          96b0e6.fade(0.93) modified.fade(0.94)
+;;     border:           000000.fade(0.82)          96b0e6.fade(0.93) modified.fade(0.8)
+;;   guide:
+;;     active:           ui.alpha(0.35)             ui.alpha(0.7)     ui.alpha(0.4)
+;;     normal:           ui.alpha(0.18)             ui.alpha(0.3)     ui.alpha(0.7)
+;;   vcs:
+;;     added:            99bf4d                     a6cc70            91b362
+;;     modified:         709ecc                     77a8d9            6994bf
+;;     removed:          f27983                     f27983            d96c75
```
I think this should be cut out

Emiller88 replied:
```suggestion
;; mirage colours:
;;  [path]      [colour] [light]                   [mirage]           [dark]
;; common:
;;   accent:    orange   ff9940                     ffcc66            e6b450
;;   bg:        black    fafafa                     1f2430            0a0e14
;;   fg:        grey     575f66                     cbccc6            b3b1ad
;;   ui:        grey     ba9199                     707abc            4d5566
;; syntax:
;;   tag:       cyan     55b4d4                     5ccfe6            39bae6
;;   func:      yellow   f2ae49                     ffd580            ffb454
;;   entity:    blue     399ee6                     73d0ff            59c2ff
;;   string:    green    86b300                     bae67e            c2d94c
;;   regexp:    teal     4cbf99                     95e6cb            95e6cb
;;   markup:    red      f07171                     f28779            f07178
;;   keyword:   orange?  fa8d3e                     ffa759            ff8f40
;;   special:   yellow?  e6ba7e                     ffe6b3            e6b673
;;   comment:   grey     abb0b6                     5c6773            626a73
;;   constant:  magenta  a37acc                     d4bfff            ffee99
;;   operator:  orange?  ed9366                     f29e74            f29668
;;   error:     red      f51818                     ff3333            ff3333
;; ui:
;;   line:               ui.alpha(0.1)              bg.darken(0.15)   bg.darken(0.2)
;;   panel:
;;     bg:               bg.brighten(0.1)           bg.brighten(0.1)  bg.brighten(0.05)
;;     shadow:           fg.darken(0.5).alpha(0.25) bg.darken(0.3)    bg.darken(0.2)
;;     border:           bg.darken(0.2)             bg.darken(0.4)    bg.darken(0.8)
;;   gutter:
;;     normal:           ui.alpha(0.4)              ui.alpha(0.4)     ui.alpha(0.6)
;;     active:           ui.alpha(0.8)              ui.alpha(0.8)     ui.alpha(0.9)
;;   selection:
;;     bg:               2ea8e6.fade(0.7)           7399e6.fade(0.87) modified.fade(0.87)
;;     inactive:         000000.fade(0.87)          96b0e6.fade(0.93) modified.fade(0.94)
;;     border:           000000.fade(0.82)          96b0e6.fade(0.93) modified.fade(0.8)
;;   guide:
;;     active:           ui.alpha(0.35)             ui.alpha(0.7)     ui.alpha(0.4)
;;     normal:           ui.alpha(0.18)             ui.alpha(0.3)     ui.alpha(0.7)
;;   vcs:
;;     added:            99bf4d                     a6cc70            91b362
;;     modified:         709ecc                     77a8d9            6994bf
;;     removed:          f27983                     f27983            d96c75
```

Emiller88 commented on themes/doom-ayu-light-theme.el:
```diff
@@ -0,0 +1,264 @@
+
+;;
+(def-doom-theme doom-ayu-light
+  "A dark theme inspired by Ayu Mirage"
```
```suggestion
  "A light theme inspired by Ayu Light"
```

Emiller88 commented on themes/doom-ayu-mirage-theme.el:
```diff
@@ -0,0 +1,263 @@
+;; mirage colours:
+;;  [path]      [colour] [light]                   [mirage]           [dark]
+;; common:
+;;   accent:    orange   ff9940                     ffcc66            e6b450
+;;   bg:        black    fafafa                     1f2430            0a0e14
+;;   fg:        grey     575f66                     cbccc6            b3b1ad
+;;   ui:        grey     ba9199                     707abc            4d5566
+;; syntax:
+;;   tag:       cyan     55b4d4                     5ccfe6            39bae6
+;;   func:      yellow   f2ae49                     ffd580            ffb454
+;;   entity:    blue     399ee6                     73d0ff            59c2ff
+;;   string:    green    86b300                     bae67e            c2d94c
+;;   regexp:    teal     4cbf99                     95e6cb            95e6cb
+;;   markup:    red      f07171                     f28779            f07178
+;;   keyword:   orange?  fa8d3e                     ffa759            ff8f40
+;;   special:   yellow?  e6ba7e                     ffe6b3            e6b673
+;;   comment:   grey     abb0b6                     5c6773            626a73
+;;   constant:  magenta  a37acc                     d4bfff            ffee99
+;;   operator:  orange?  ed9366                     f29e74            f29668
+;;   error:     red      f51818                     ff3333            ff3333
+;; ui:
+;;   line:               ui.alpha(0.1)              bg.darken(0.15)   bg.darken(0.2)
+;;   panel:
+;;     bg:               bg.brighten(0.1)           bg.brighten(0.1)  bg.brighten(0.05)
+;;     shadow:           fg.darken(0.5).alpha(0.25) bg.darken(0.3)    bg.darken(0.2)
+;;     border:           bg.darken(0.2)             bg.darken(0.4)    bg.darken(0.8)
+;;   gutter:
+;;     normal:           ui.alpha(0.4)              ui.alpha(0.4)     ui.alpha(0.6)
+;;     active:           ui.alpha(0.8)              ui.alpha(0.8)     ui.alpha(0.9)
+;;   selection:
+;;     bg:               2ea8e6.fade(0.7)           7399e6.fade(0.87) modified.fade(0.87)
+;;     inactive:         000000.fade(0.87)          96b0e6.fade(0.93) modified.fade(0.94)
+;;     border:           000000.fade(0.82)          96b0e6.fade(0.93) modified.fade(0.8)
+;;   guide:
+;;     active:           ui.alpha(0.35)             ui.alpha(0.7)     ui.alpha(0.4)
+;;     normal:           ui.alpha(0.18)             ui.alpha(0.3)     ui.alpha(0.7)
+;;   vcs:
+;;     added:            99bf4d                     a6cc70            91b362
+;;     modified:         709ecc                     77a8d9            6994bf
+;;     removed:          f27983                     f27983            d96c75
```
I like this because the Ayu theme was really difficult to port because of its complexity(using slightly different shades for stuff). I don't know if I like it in the theme file, since none of the other themes have this(to my knowledge).

LoveSponge replied:
yeah i completely agree, i just committed what i had at one stage, completely understanding that it's still WIP, in need of editing - i left these particular comments in as a 'source of reference' for each theme when i came to copy/paste the files for the different variants. Definitely agree they should be removed though.

Emiller88 replied:
```suggestion
;; mirage colours:
;;  [path]      [colour] [light]                   [mirage]           [dark]
;; common:
;;   accent:    orange   ff9940                     ffcc66            e6b450
;;   bg:        black    fafafa                     1f2430            0a0e14
;;   fg:        grey     575f66                     cbccc6            b3b1ad
;;   ui:        grey     ba9199                     707abc            4d5566
;; syntax:
;;   tag:       cyan     55b4d4                     5ccfe6            39bae6
;;   func:      yellow   f2ae49                     ffd580            ffb454
;;   entity:    blue     399ee6                     73d0ff            59c2ff
;;   string:    green    86b300                     bae67e            c2d94c
;;   regexp:    teal     4cbf99                     95e6cb            95e6cb
;;   markup:    red      f07171                     f28779            f07178
;;   keyword:   orange?  fa8d3e                     ffa759            ff8f40
;;   special:   yellow?  e6ba7e                     ffe6b3            e6b673
;;   comment:   grey     abb0b6                     5c6773            626a73
;;   constant:  magenta  a37acc                     d4bfff            ffee99
;;   operator:  orange?  ed9366                     f29e74            f29668
;;   error:     red      f51818                     ff3333            ff3333
;; ui:
;;   line:               ui.alpha(0.1)              bg.darken(0.15)   bg.darken(0.2)
;;   panel:
;;     bg:               bg.brighten(0.1)           bg.brighten(0.1)  bg.brighten(0.05)
;;     shadow:           fg.darken(0.5).alpha(0.25) bg.darken(0.3)    bg.darken(0.2)
;;     border:           bg.darken(0.2)             bg.darken(0.4)    bg.darken(0.8)
;;   gutter:
;;     normal:           ui.alpha(0.4)              ui.alpha(0.4)     ui.alpha(0.6)
;;     active:           ui.alpha(0.8)              ui.alpha(0.8)     ui.alpha(0.9)
;;   selection:
;;     bg:               2ea8e6.fade(0.7)           7399e6.fade(0.87) modified.fade(0.87)
;;     inactive:         000000.fade(0.87)          96b0e6.fade(0.93) modified.fade(0.94)
;;     border:           000000.fade(0.82)          96b0e6.fade(0.93) modified.fade(0.8)
;;   guide:
;;     active:           ui.alpha(0.35)             ui.alpha(0.7)     ui.alpha(0.4)
;;     normal:           ui.alpha(0.18)             ui.alpha(0.3)     ui.alpha(0.7)
;;   vcs:
;;     added:            99bf4d                     a6cc70            91b362
;;     modified:         709ecc                     77a8d9            6994bf
;;     removed:          f27983                     f27983            d96c75
```

LoveSponge commented:
thanks for reviewing!
I'll make the changes you suggested, i am going to remove all the comments you mentioned and split them out into a (non-committed) file for future reference.

There is currently an issue with the mirage theme in the company suggestions window - it seems to be using the wrong colour but i haven't been able to work out why yet... Your help on this would be greatly appreciated.

Also worth mentioning is some code at the bottom which refers specifcally to RJSX-mode speficic highlight rules, i'm not sure if this will break if a given user does not have RJSX-mode installed/activated.

Cheers,
LS

Emiller88 commented:
@LoveSponge I've been using ayu-light for about a week and have no complaints, besides that the yellow is sometimes too light with the background but I didn't write down when I ran into that.

I'll be using mirage this week. So far I found the `DONE` org-mode items are the same as the bg so they disappear.

LoveSponge commented:
@Emiller88 Thank you! i hope you are enjoying it for the most part =)

I need to give this a little bit of love but i don't really know much about authoring emacs themes, let alone doom themes. Some help with the specific display problems would be very helpful!

I personally use the mirage theme full-time; It works mostly perfectly but there is an issue with the 'company' (suggestion box thing) box where it shows up white with white text, making it very difficult to see suggestions, not sure what caused this as it seems to be the only thing that has this problem.

Thanks again for checking out my work =)

LoveSponge commented:
@ema2159:
>  other PR is also ok because those screenshots will also be merged when the themes

Just to be clear, you don't need the screenshots pasted here as they are referenced in the other PR?

ema2159 commented:
@LoveSponge no! It's ok like that! By the way, very pretty looking themes. I saw @Emiller88 took the task of reviewing it, so I'm not sure if he needs any help.

LoveSponge commented:
Sorry this took me so long guys, i really appreciate the comments and help on this.

Please check out my comments above about known issues.

Thanks again!

LoveSponge commented:
Didn't mean to hit close. #n00b..

Emiller88 commented:
Looks good to me! There may be a few formatting things to fix, but I requested @ema2159 and @hlissner make sure we follow the formatting.

ema2159 commented on themes/doom-ayu-light-theme.el:
```diff
@@ -0,0 +1,224 @@
+determine the exact padding."
+  :group 'doom-ayu-light-theme
+  :type '(choice integer boolean))
+
```
I'd ask you to fix these spaces and the ones inline 37 and 41.

ema2159 commented on themes/doom-ayu-mirage-theme.el:
```diff
@@ -0,0 +1,223 @@
+   (rjsx-tag-bracket-face :foreground (doom-darken cyan 0.5))
+   (rjsx-attr :foreground syntax-func)
+  )
+ ;; '(rjsx-tag ((((class color) (min-colors 257)) (:background "#0a0e14" :foreground "#39bae6")) (((class color) (min-colors 256)) (:background nil :foreground "#bfbfbf")) (((class color) (min-colors 16)) (:background nil :foreground "brightwhite"))))
```
Remove these comments (and all the commented code in the themes).

ema2159 commented:
@LoveSponge I left some reviews. Also, I'd like to ask you to check company's faces in Ayu Mirage since for some reason the same faces as Ayu light are being displayed.

LoveSponge commented:
Updated mirage theme, still not completely done, but it's much closer now to where i would like it to be.
Light theme probably still needs some work.

I've addressed your reviews and comments:
- commented code
- empty lines
- org mode highlighting
- company popup highlight

and these extra tidbits
- numbers being bold
- overall font colour corrected
- incorrect hex code corrected (common-ui)
- ivy (match highlights, overall colour)
- line highlight
- line selection highlight (`doom-blend`! what a blessing! I'll probably find more places to use this now that I know how to use it.)

Let me know, thanks in advance.

LoveSponge commented:
Update:

Brought light theme into line with new mirage changes.
Removed some unneeded customisations.

LoveSponge commented:
Update:
Fixed line highlight
Fixed treemacs line highlight
Fixed ivy highlight

Minor changes and improvements, still waiting for this to be merged...

benbovy commented:
I'd love to use this theme when available!

ema2159 commented:
Great job man!
